tikv/raftstore/engine.go

```diff
@@ -3,9 +3,17 @@ package raftstore
 import (
 	"github.com/coocood/badger"
 	"github.com/golang/protobuf/proto"
+	"github.com/ngaut/unistore/lockstore"
 	"github.com/pingcap/errors"
 )
 
+type DBSnapshot struct {
+	db            *badger.DB
+	txn           *badger.Txn
+	lockStore     *lockstore.MemStore
```

Comment:
Do we need to keep the consistency between badger.DB and MemStore?

Reply:
Yes, we should handle it later.